torch/_inductor/kernel/custom_op.py

```diff
@@ -318,20 +321,70 @@ def autotune_custom_op(
         )
         input_gen_fns = _adapt_user_input_gen_fns(inputs, arg_names, user_input_gen_fns)
 
-    return autotune_select_algorithm(
+    # Run autotuning to select the best choice
+    if not enable_fusion:
+        return autotune_select_algorithm(
+            name=name,
+            choices=choices,
+            input_nodes=list(inputs),
+            layout=choices[0].layout,
+            input_gen_fns=input_gen_fns,
+        )
+
+    # Request both the result and the winning choice when fusion is enabled
+    selected_result, winning_choice = autotune_select_algorithm(
         name=name,
         choices=choices,
         input_nodes=list(inputs),
         layout=choices[0].layout,
         input_gen_fns=input_gen_fns,
+        return_choice=True,
     )
 
+    # Apply inlining for fusion if winning_choice has graph; otherwise return result as-is(default impl)
+    if hasattr(winning_choice, "gm"):
```

Comment:
We shouldn't be conditionally assigning attrs. can this be None if it's not set ?